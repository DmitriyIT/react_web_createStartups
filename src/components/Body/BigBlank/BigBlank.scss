.BigBlank {
	width: 581px;
	overflow: hidden;
	border-radius: 10px 10px 5px 5px;
	box-shadow: 1px 4px 20px rgba(0, 0, 0, 0.15);

	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	font-size: 15px;

	color: #345761;
	
	.BigBlank__title {
		min-height: 50px;
		padding-left: 20px;
		width: 100%;
		background: linear-gradient(165.66deg, #FFE99C 16.29%, #FFE486 115.92%, rgba(255, 164, 91, 0.8) 166.56%);
		line-height: 50px;
		color: #24505F;
	}
	.BigBlank_err {
		color: rgba(207, 88, 88, 1);
		margin-left: 20px;
		margin-top: 10px;
		margin-bottom: -3px;
	}
	form {
		padding: 22px;
		font-size: 14px;
		color: #345761;
		max-height: 345px;
		overflow-x: scroll;

		.BigBlank__field {
			padding: 10px;
			width: 90%;
			max-width: 92%;
			margin-bottom: 10px;
			margin-top: 5px;
			background: #F3F3F3;
			border: none;
			border-radius: 5px;

	
			font-family: Fira Sans;
			font-style: italic;
			font-weight: 500;
			font-size: 14px;

			color: #596467;
		}
	}
	.BigBlank__button {
		cursor: pointer;
		float: right;
		margin: 10px 20px 20px 0;
		min-height: 35px;
		padding-left: 20px;
		padding-right: 20px;
		background: linear-gradient(160.87deg, #FFE486 145.71%, #009D90 431.65%);
		border: none;

		line-height: 40px;
		font-weight: 500;
		font-size: 15px;
		color: #265D6E;
	}
}
