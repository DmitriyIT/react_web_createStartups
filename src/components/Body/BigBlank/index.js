import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './BigBlank.scss';

import AnimateCreate from '../AnimateCreate.js';

function BigBlank({ err, title, fields_input, button_left, button_text, submitForm, textareaChange }) {
	
	var fields = fields_input.map(e => [
		<div className="BigBlank__titleField">{e.text}</div>,
		<textarea className="BigBlank__field" name={e.name} value={e.value} onChange={textareaChange} />
	]);
	return (
		<div className="BigBlank">
			<div className="BigBlank__title">{title}</div>
			{ err ? <div className="BigBlank_err">{err}</div> : ''}
			<AnimateCreate>
			<form onSubmit={submitForm} id="BigBlank">
				{fields}
			</form>
			</AnimateCreate>

			<input type="submit" form="BigBlank" className="BigBlank__button" value={button_text} />
			
			{button_left ? (
				<Link to={button_left.link}>
					<div className="BigBlank__button"> {button_left.text} </div>
				</Link>
			) : ''}
			
		</div>
	)
}

export default BigBlank;