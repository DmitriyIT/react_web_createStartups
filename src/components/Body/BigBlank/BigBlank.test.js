import React, { Component } from 'react';
import { Link, BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import BigBlank from './index.js';

var props = {
	err: 'awd',
	title: 'заголовок',
	fields_input: [
		{name: 'theme', text: "Тема стартапа", value: ''}, 
		{name: 'description', text: "Описание подробное", value: ''}
	],
	button_left: {
		link: "/mystartup/main",
		text: "отмена"
	},
	button_text: 'создать',
	submitForm: () => {},
	textareaChange: () => {}
}

it('base path, all include', () => {
	var component = renderer.create(
		<BrowserRouter>
			<BigBlank {...props} />
		</BrowserRouter>
	);
	expect(component.toJSON()).toMatchSnapshot();
})

it('without buttonLeft', () => {
	delete(props.button_left);
	var component = renderer.create(
		<BrowserRouter>
			<BigBlank {...props} />
		</BrowserRouter>
	);
	expect(component.toJSON()).toMatchSnapshot();
})

it('without buttonLeft && error', () => {
	delete(props.err);
	var component = renderer.create(
		<BrowserRouter>
			<BigBlank {...props} />
		</BrowserRouter>
	);
	expect(component.toJSON()).toMatchSnapshot();
})