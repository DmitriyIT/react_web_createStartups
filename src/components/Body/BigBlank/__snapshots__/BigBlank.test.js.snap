// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base path, all include 1`] = `
<div
  className="BigBlank"
>
  <div
    className="BigBlank__title"
  >
    заголовок
  </div>
  <div
    className="BigBlank_err"
  >
    awd
  </div>
  <form
    id="BigBlank"
    onSubmit={[Function]}
  >
    <div
      className="BigBlank__titleField"
    >
      Тема стартапа
    </div>
    <textarea
      className="BigBlank__field"
      name="theme"
      onChange={[Function]}
      value=""
    />
    <div
      className="BigBlank__titleField"
    >
      Описание подробное
    </div>
    <textarea
      className="BigBlank__field"
      name="description"
      onChange={[Function]}
      value=""
    />
  </form>
  <input
    className="BigBlank__button"
    form="BigBlank"
    type="submit"
    value="создать"
  />
  <a
    href="/mystartup/main"
    onClick={[Function]}
  >
    <div
      className="BigBlank__button"
    >
       
      отмена
       
    </div>
  </a>
</div>
`;

exports[`without buttonLeft && error 1`] = `
<div
  className="BigBlank"
>
  <div
    className="BigBlank__title"
  >
    заголовок
  </div>
  
  <form
    id="BigBlank"
    onSubmit={[Function]}
  >
    <div
      className="BigBlank__titleField"
    >
      Тема стартапа
    </div>
    <textarea
      className="BigBlank__field"
      name="theme"
      onChange={[Function]}
      value=""
    />
    <div
      className="BigBlank__titleField"
    >
      Описание подробное
    </div>
    <textarea
      className="BigBlank__field"
      name="description"
      onChange={[Function]}
      value=""
    />
  </form>
  <input
    className="BigBlank__button"
    form="BigBlank"
    type="submit"
    value="создать"
  />
  
</div>
`;

exports[`without buttonLeft 1`] = `
<div
  className="BigBlank"
>
  <div
    className="BigBlank__title"
  >
    заголовок
  </div>
  <div
    className="BigBlank_err"
  >
    awd
  </div>
  <form
    id="BigBlank"
    onSubmit={[Function]}
  >
    <div
      className="BigBlank__titleField"
    >
      Тема стартапа
    </div>
    <textarea
      className="BigBlank__field"
      name="theme"
      onChange={[Function]}
      value=""
    />
    <div
      className="BigBlank__titleField"
    >
      Описание подробное
    </div>
    <textarea
      className="BigBlank__field"
      name="description"
      onChange={[Function]}
      value=""
    />
  </form>
  <input
    className="BigBlank__button"
    form="BigBlank"
    type="submit"
    value="создать"
  />
  
</div>
`;
